src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/CompleteGoalTest.kt

```diff
@@ -0,0 +1,368 @@
+  }
+
+  @Test
+  fun `should return 409 if goal still has active or not started steps`() {
+    // given
+    val goalReference = createAGoalAndGetTheReference(prisonNumber)
+    val completeGoalRequest = aValidCompleteGoalRequest(
+      goalReference = goalReference,
+    )
+    // when
+    val response = completeAGoal(prisonNumber, goalReference, completeGoalRequest)
+      .expectStatus()
+      .isEqualTo(HttpStatus.CONFLICT)
+      .returnResult(ErrorResponse::class.java)
+
+    // then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(HttpStatus.CONFLICT.value())
+      .hasUserMessageContaining("Could not complete goal with reference [$goalReference] for prisoner [$prisonNumber]: As it has active or not started steps")
+  }
+
+  @Test
+  fun `should return 409 if goal is already completed`() {
+    // given
+    val goalReference = createAGoalAndGetTheReference(prisonNumber)
+    val completeGoalRequest = aValidCompleteGoalRequest(goalReference = goalReference)
+    completeSteps(prisonNumber)
+    completeAGoal(prisonNumber, goalReference, completeGoalRequest)
+      .expectStatus()
+      .isNoContent()
+
+    // when
+    val response = completeAGoal(prisonNumber, goalReference, completeGoalRequest)
+      .expectStatus()
+      .isEqualTo(HttpStatus.CONFLICT)
+      .returnResult(ErrorResponse::class.java)
+
+    // then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(HttpStatus.CONFLICT.value())
+      .hasUserMessageContaining("Could not complete goal with reference [$goalReference] for prisoner [$prisonNumber]: Goal was in state [COMPLETED] that can't be completed")
+  }
+
+  private fun completeAGoal(
+    prisonNumber: String,
+    goalReference: UUID,
+    completeGoalRequest: CompleteGoalRequest,
+  ): WebTestClient.ResponseSpec = webTestClient.put()
+    .uri(URI_TEMPLATE, prisonNumber, goalReference)
+    .withBody(completeGoalRequest)
+    .bearerToken(
+      aValidTokenWithAuthority(
+        GOALS_RW,
+        username = "buser_gen",
+        displayName = "Bernie User",
+        privateKey = keyPair.private,
+      ),
+    )
+    .contentType(APPLICATION_JSON)
+    .exchange()
+
+  private fun createAGoalAndGetTheReference(prisonNumber: String): UUID {
+    val createGoalRequest = aValidCreateGoalRequest(
+      title = "Learn French",
+      steps = listOf(
+        aValidCreateStepRequest(
+          title = "Book course",
+        ),
+        aValidCreateStepRequest(
+          title = "Attend course",
+        ),
+      ),
+    )
+    createGoal(
+      username = "auser_gen",
+      displayName = "Albert User",
+      prisonNumber = prisonNumber,
+      createGoalRequest = createGoalRequest,
+    )
+    val actionPlan = getActionPlan(prisonNumber)
+    val goal = actionPlan.goals[0]
+    val goalReference = goal.goalReference
+    return goalReference
+  }
+
+  fun completeSteps(prisonNumber: String) {
```

Comment:
Added this method to complete the child goal steps - may need this elsewhere at some point so is a candidate for adding to the super class ?.